### PR TITLE
Upgrade to Kotlin 1.3

### LINKED ISSUE #35: Move to Kotlin 1.3
AC: Summary

Blocks:
* #115 
* #113

### PR DESCRIPTION
**Pull request by zach-klippenstein**
This branch has 2 commits:
1. Upgrade the actual dep, fix the imports, and then fix bugs due to changes in error handling behavior.
2. Fix experimental and deprecation warnings (mostly by just suppressing them).

This PR targets the `kotlin-13` branch, which is still a bit old. I want to land on that branch first, before merging master, so that I can upgrade the rewritten workflows code separately.

Closes #35.

### CODE REVIEW

File: kotlin/legacy/legacy-workflow-rx2/src/test/java/com/squareup/workflow/legacy/rx2/WorkflowOperatorsTest.kt [hunk 1]
```diff
@@ -77,6 +77,8 @@ class WorkflowOperatorsTest {
     states.cancel()
 
     assertThat(subscribeCount).isEqualTo(1)
-    assertThat(disposeCount).isEqualTo(1)
+    // For some reason the observable is actually disposed twice. Seems like a coroutines bug, but
+    // Disposable.dispose() is an idempotent operation so it should be fine.
```
Comment (kirillzh) [inline kotlin/legacy/legacy-workflow-rx2/src/test/java/com/squareup/workflow/legacy/rx2/WorkflowOperatorsTest.kt]: Interesting... I assume that's regression, worth filing Kotlin bug?

Comment (zach-klippenstein) [reply kotlin/legacy/legacy-workflow-rx2/src/test/java/com/squareup/workflow/legacy/rx2/WorkflowOperatorsTest.kt]: I believe so. I'll try to isolate and file this week.

File: kotlin/legacy/legacy-workflow-rx2/src/main/java/com/squareup/workflow/legacy/rx2/EventSelectBuilder.kt [hunk 2]
```diff
@@ -77,6 +78,11 @@ class EventSelectBuilder<E : Any, R : Any> internal constructor(
    */
   @PublishedApi
   internal val scope = CoroutineScope(Unconfined + selectionJob)
+      .apply {
+        coroutineContext[Job]!!.invokeOnCompletion { cause ->
+          cause?.printStackTrace()
```
Comment (rjrjr) [inline kotlin/legacy/legacy-workflow-rx2/src/main/java/com/squareup/workflow/legacy/rx2/EventSelectBuilder.kt]: Sketchy in production code?

File: kotlin/legacy/legacy-workflow-core/src/main/java/com/squareup/workflow/legacy/WorkflowOperators.kt [hunk 3]
```diff
@@ -113,6 +115,7 @@ fun <S : Any, E : Any, O1 : Any, O2 : Any> Workflow<S, E, O1>.mapResult(
   // Propagate cancellation upstream.
   transformedResult.invokeOnCompletion { cause ->
     if (cause != null) {
+      @Suppress("DEPRECATION")
```
Comment (rjrjr) [inline kotlin/legacy/legacy-workflow-core/src/main/java/com/squareup/workflow/legacy/WorkflowOperators.kt]: Worth noting why we can't use the un-deprecated path, if there is one.

File: kotlin/workflow-testing/src/test/java/com/squaruep/workflow/ChannelSubscriptionsIntegrationTest.kt [hunk 4]
```diff
@@ -165,6 +167,7 @@ class ChannelSubscriptionsIntegrationTest {
       workflow.testFromStart(true) { host ->
         assertFalse(host.hasOutput)
 
+        @Suppress("DEPRECATION")
```
Comment (rjrjr) [inline kotlin/workflow-testing/src/test/java/com/squaruep/workflow/ChannelSubscriptionsIntegrationTest.kt]: Another spot for a note about why we have to use deprecated.

File: kotlin/legacy/legacy-workflow-rx2/src/main/java/com/squareup/workflow/legacy/rx2/EventSelectBuilder.kt (hunk 2 above)
Comment (zach-klippenstein) [reply kotlin/legacy/legacy-workflow-rx2/src/main/java/com/squareup/workflow/legacy/rx2/EventSelectBuilder.kt]: Oops, didn't mean to leave this in.

File: kotlin/legacy/legacy-workflow-core/src/main/java/com/squareup/workflow/legacy/WorkflowOperators.kt (hunk 3 above)
Comment (zach-klippenstein) [reply kotlin/legacy/legacy-workflow-core/src/main/java/com/squareup/workflow/legacy/WorkflowOperators.kt]: Filed #188 and added TODOs.

**Pull request merged**